### PR TITLE
gpavelar/c03-make01

### PR DESCRIPTION
**Pull request by gpavelar**
# Description

Exercise (e.g. C01-E01): C03-MAKE01

Student Name: Gustavo Avelar

> Create Makefile to interact with Docker container using simple commands.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [x] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/03class/exercises/c03-make01/gpavelar/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM httpd:2.4.41-alpine
+ARG REGISTRY
+ENV REGISTRY ${REGISTRY}
+ARG TAG
+ENV TAG ${TAG}
```
Comment (alanlima) [inline classes/03class/exercises/c03-make01/gpavelar/Dockerfile:5]: You don't need to specify ENV variables in this case. Those variables will not be changed after image is built.

- **ARG:** variables that can be modified at building time only. via `docker build --build-args`
- **ENV:** variables that can be preset at building time and can be modified at runtime. via `docker run --env`

This image might help to understand:

Docker ARG vs ENV

File: classes/03class/exercises/c03-make01/gpavelar/Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	@echo "Building Docker Image ${REGISTRY}/${IMAGE}:${TAG}"
+	@docker build -t ${IMAGE}:${TAG} \
+		--build-arg REGISTRY=${REGISTRY} \
+		--build-arg TAG=${TAG} .
+
+.PHONY: run
+run:
+	@echo "Running the container on port 8081..."
+	@docker run -d -p 8081:80 ${IMAGE}:${TAG}
```
Comment (alanlima) [inline classes/03class/exercises/c03-make01/gpavelar/Makefile:19]: The port should be passed as parameter here:

> The port to be mapped for the container should be dynamically provided during the make run command.

File: classes/03class/exercises/c03-make01/gpavelar/Dockerfile (hunk 1 above)
Comment (gpavelar) [reply classes/03class/exercises/c03-make01/gpavelar/Dockerfile:5]: @alanlima Thanks alot, I really understood it in a different way. I am getting used to it. So I have save this on my notes, but i will need some time to make those changes.

File: classes/03class/exercises/c03-make01/gpavelar/Makefile (hunk 2 above)
Comment (gpavelar) [reply classes/03class/exercises/c03-make01/gpavelar/Makefile:19]: oh no, I will change it.

**Review by alanlima — APPROVED** [review]
Thanks for the changes @gpavelar

Nice work 🚀

**Pull request merged**